### PR TITLE
Add utility functions to access various part of source capabilities

### PR DESCRIPTION
**Pull request by sjoerdsimons**

### CODE REVIEW

File: usb-pd/src/messages/pdo.rs [hunk 1]
```diff
@@ -102,11 +141,25 @@ bitfield! {
...
-        pub min_voltage: u8 @ 8..=15,
+        pub raw_min_voltage: u8 @ 8..=15,
         /// Maximum Current in 50mA increments
-        pub maximum_current: u8 @ 0..=6,
+        pub raw_max_current: u8 @ 0..=6,
+    }
+}
+
+impl SPRProgrammablePowerSupply {
+    pub fn max_voltage_mv(&self) -> u16 {
```
Comment (fmckeogh) [inline usb-pd/src/messages/pdo.rs:153]: I understand this is subjective, but I'm not loving the `_mv` here. Which of the following do you think is best?

* `max_voltage() -> u16`, with a doc comment explaining the return value is in millivolts
* `max_voltage() -> MilliVolts` where `MilliVolts` is some type representing voltage units (I'm assuming there must be some crate with these types already)
* `max_voltage_millivolts() -> u16`

File: usb-pd/src/messages/pdo.rs [hunk 2]
```diff
@@ -119,11 +172,21 @@ bitfield! {
...
+        pub raw_min_voltage: u8 @ 8..=15,
         /// PDP in 1W increments
-        pub maximum_current: u8 @ 0..=7,
+        pub pd_power: u8 @ 0..=7,
+    }
+}
+
+impl EPRAdjustableVoltageSupply {
+    pub fn max_voltage_mv(&self) -> u16 {
+        u16::from(self.raw_max_voltage()) * 100
```
Comment (fmckeogh) [inline usb-pd/src/messages/pdo.rs:185]: Unnecessary `u16::from` here?

File: usb-pd/src/messages/pdo.rs (hunk 1 above)
Comment (sjoerdsimons) [reply usb-pd/src/messages/pdo.rs:153]: There is `uom` which is meant to have no_std functionality and has types for e.g. voltage; So i can have a look at that to see if that ends up being overkill or note.. 

fwiw I discounted the first option is it ends up being surprising if voltage doesn't give you volts.. The third  option is also ok by me, it's just a bit more typing (with mv being a common shorthand)

File: usb-pd/src/messages/pdo.rs (hunk 2 above)
Comment (sjoerdsimons) [reply usb-pd/src/messages/pdo.rs:185]: indeed!

File: usb-pd/src/messages/pdo.rs (hunk 1 above)
Comment (fmckeogh) [reply usb-pd/src/messages/pdo.rs:153]: I like `uom`, I'm happy to merge this commit as is, and I'll implement it.

**Pull request merged**